plugins/inputs/t128_tank/t128_tank.go

```diff
@@ -99,7 +99,7 @@ func (plugin *T128Tank) Start(acc telegraf.Accumulator) error {
 		plugin.adjustTime = func(m telegraf.Metric) {
 			mTime := m.Time()
 			if mTime.Before(unreasonableTimestamp) {
-				adjustedTime := unreasonableTimestamp.Unix()
+				adjustedTime := unreasonableTimestamp.Unix() * int64(plugin.Precision)
```

Comment:
This still isn't right. Why are you using `unreasonableTimestamp`? It's an "unreasonable timestamp"...You need to replicate the logic from the time adjustment in `init`.

Reply:
I was using that because the unreasonable time would be 1970 or something... but mtime would be sometime before that so wouldn't it make sense to set time to 1970?

Reply:
You're missing the point. If the time is this small, it means it was incorrectly interpreted as nanoseconds. The timestamp is actually in seconds and the nanoseconds->seconds conversion should be performed on the timestamp we received as if the user had configured precision=1s.